DataFormats/Candidate/interface/LeafCandidate.h

```diff
@@ -33,6 +33,8 @@ namespace reco {
 
     typedef unsigned int index;
 
+    LeafCandidate() {}
```

Comment:
(I typed in my last comment https://github.com/cms-sw/cmssw/pull/7097#discussion_r22731446  before Bill's one https://github.com/cms-sw/cmssw/pull/7097#discussion_r22731213 showed up)
it sounds like I got it about right.